internal/cmd/gocdk: init should create a go.sum file

### Describe the bug

`gocdk init` does not create a go.sum file when first creating the project. This causes problems when running `gocdk build`, because the Dockerfile expects the file to exists.

### To Reproduce

1. `gocdk init foo`
2. `cd foo`
3. `gocdk build` (this will crash)

### Expected behavior

Build works, but perhaps not verifiable. (I think an empty go.sum would likely be fine.)

### Version

master (1d5466654942d3ec020d239e97666bc5fe60acaa)

What's the down side of just running `go build` as part of `gocdk build`? IIUC, it would create the `go.sum` file (so `gocdk init` wouldn't have to change), and it also helps identify compile errors much more quickly. E.g.,:

1. Create a project from scratch (`gocdk init foo && cd foo`)
2. `touch go.sum` (to work around this bug)
3. Edit `main.go` to introduce a compile error
4. `time gocdk build`

-->

```
# foo
./main.go:17:2: undefined: xhttp
The command '/bin/sh -c go build -o foo' returned a non-zero code: 2
Error: gocdk build: docker build: exit status 2
user	0m1.407s
```

vs

1. Create a project from scratch (`gocdk init foo && cd foo`)
2. Edit `main.go` to introduce a compile error
3. `time go build`

-->
 
```
# foo
./main.go:17:2: undefined: xhttp
user	0m0.621s
```

(and `go.sum` now exists).

--- a/internal/cmd/gocdk/build.go
+++ b/internal/cmd/gocdk/build.go
@@ -86,6 +86,15 @@ func build(ctx context.Context, pctx *processContext, refs []string) (string, er
 	if err != nil {
 		return "", xerrors.Errorf("gocdk build: %w", err)
 	}
+	// Run a local build to identify compile errors, etc.
+	// TODO(rvangent): Consider using go/packages instead, to avoid the cost of linking.
+	pctx.Logf("Running \"go build\"...")
+	buildCmd := pctx.NewCommand(ctx, moduleRoot, "go", "build")
+	buildCmd.Env = append(buildCmd.Env, "GO111MODULE=on")
+	if err := buildCmd.Run(); err != nil {
+		return "", xerrors.Errorf("gocdk build: go build: %w", err)
+	}
+	pctx.Logf("\"go build\" succeeded.")
 	var imageName string
 	for i := range refs {
 		if !strings.HasPrefix(refs[i], ":") {

